libtenzir/builtins/formats/json.cpp

```diff
@@ -1193,10 +1193,15 @@ class json_parser final : public plugin_parser {
     return "json";
   }
 
-  auto optimize(event_order order) -> std::unique_ptr<plugin_parser> override {
+  auto optimize(expression const& filter, event_order order,
+                select_optimization const& selection)
+    -> optimize_parser_result override {
+    (void)filter;
+    (void)selection;
     auto args = args_;
     args.preserve_order = order == event_order::ordered;
-    return std::make_unique<json_parser>(std::move(args));
+    return {std::make_unique<json_parser>(std::move(args)), false, false,
+            false};
```

Comment:
As far as I understand, one of the `false, false, false` indicates that the order was not optimized. But it was optimized right above this line.